pkg/pxc/backup/restore.go

```diff
@@ -227,6 +227,157 @@ func PVCRestoreJob(cr *api.PerconaXtraDBClusterRestore, cluster api.PerconaXtraD
 	return job, nil
 }
 
+func AzureRestoreJob(cr *api.PerconaXtraDBClusterRestore, bcp *api.PerconaXtraDBClusterBackup, cluster api.PerconaXtraDBClusterSpec, destination string) (*batchv1.Job, error) {
+	if bcp.Status.Azure == nil {
+		return nil, errors.New("nil azure backup status")
+	}
+
+	jobPVC := corev1.Volume{
+		Name: "datadir",
+		VolumeSource: corev1.VolumeSource{
+			PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+				ClaimName: "datadir-" + cr.Spec.PXCCluster + "-pxc-0",
+			},
+		},
+	}
+
+	jobPVCs := []corev1.Volume{
+		jobPVC,
+		app.GetSecretVolumes("vault-keyring-secret", cluster.PXC.VaultSecretName, true),
+	}
+	pxcUser := "xtrabackup"
+	command := []string{"recovery-azure.sh"}
+
+	verifyTLS := true
+	if cluster.Backup != nil && len(cluster.Backup.Storages) > 0 {
+		storage, ok := cluster.Backup.Storages[bcp.Spec.StorageName]
+		if ok && storage.VerifyTLS != nil {
+			verifyTLS = *storage.VerifyTLS
+		}
+	}
+	azure := bcp.Status.Azure
+	storageAccount := corev1.EnvVar{
+		Name: "AZURE_STORAGE_ACCOUNT",
+		ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: app.SecretKeySelector(azure.CredentialsSecret, "AZURE_STORAGE_ACCOUNT"),
+		},
+	}
+	accessKey := corev1.EnvVar{
+		Name: "AZURE_ACCESS_KEY",
+		ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: app.SecretKeySelector(azure.CredentialsSecret, "AZURE_ACCESS_KEY"),
+		},
+	}
+	containerName := corev1.EnvVar{
+		Name:  "AZURE_CONTAINER_NAME",
+		Value: azure.ContainerName,
+	}
+	endpoint := corev1.EnvVar{
+		Name:  "AZURE_ENDPOINT",
+		Value: azure.Endpoint,
+	}
+	storageClass := corev1.EnvVar{
+		Name:  "AZURE_STORAGE_CLASS",
+		Value: azure.StorageClass,
+	}
+	backupPath := corev1.EnvVar{
+		Name:  "BACKUP_PATH",
+		Value: strings.TrimPrefix(destination, azure.ContainerName+"/"),
+	}
+	envs := []corev1.EnvVar{storageAccount, accessKey, containerName, endpoint, storageClass, backupPath,
```

Comment:
it looks really strange, why do we define some EnvVars separately and the others in slice itself?